src/stan/analyze/mcmc/compute_potential_scale_reduction.hpp

```diff
@@ -17,156 +17,6 @@
 namespace stan {
 namespace analyze {
 
-/**
- * Computes normalized average ranks for draws. Transforming them to normal
- * scores using inverse normal transformation and a fractional offset. Based on
- * paper https://arxiv.org/abs/1903.08008
- * @param chains stores chains in columns
- * @return normal scores for average ranks of draws
- */
-inline Eigen::MatrixXd rank_transform(const Eigen::MatrixXd& chains) {
-  const Eigen::Index rows = chains.rows();
-  const Eigen::Index cols = chains.cols();
-  const Eigen::Index size = rows * cols;
-
-  std::vector<std::pair<double, int>> value_with_index(size);
-
-  for (Eigen::Index i = 0; i < size; ++i) {
-    value_with_index[i] = {chains(i), i};
-  }
-
-  std::sort(value_with_index.begin(), value_with_index.end());
-
-  Eigen::MatrixXd rank_matrix = Eigen::MatrixXd::Zero(rows, cols);
-
-  // Assigning average ranks
-  for (Eigen::Index i = 0; i < size; ++i) {
-    // Handle ties by averaging ranks
-    Eigen::Index j = i + 1;
-    double sum_ranks = j;
-    Eigen::Index count = 1;
-
-    while (j < size && value_with_index[j].first == value_with_index[i].first) {
-      sum_ranks += j + 1;  // Rank starts from 1
-      ++j;
-      ++count;
-    }
-    double avg_rank = sum_ranks / count;
-    boost::math::normal_distribution<double> dist;
-    for (std::size_t k = i; k < j; ++k) {
-      double p = (avg_rank - 3.0 / 8.0) / (size - 2.0 * 3.0 / 8.0 + 1.0);
-      const Eigen::Index index = value_with_index[k].second;
-      rank_matrix(index) = boost::math::quantile(dist, p);
-    }
-    i = j - 1;  // Skip over tied elements
-  }
-  return rank_matrix;
-}
-
-/**
- * Computes square root of marginal posterior variance of the estimand by the
- * weigted average of within-chain variance W and between-chain variance B.
- *
- * @param chains stores chains in columns
- * @return square root of ((N-1)/N)W + B/N
- */
-inline double rhat(const Eigen::MatrixXd& chains) {
```

Comment:
these haven't been deleted, they've just been moved to file `split_rank_normalized_rhat`.
this was done for 2 reasons: 
1. so that `rank_transform` can be used by split_rank_normalized_ess as well
2. the Stan modules sometimes have 1 function per file and sometimes are a whole set of functions in a file. when the functions are as complex as computing Rhat and ESS, the fewer functions per file, the easier it is for me to deal with them.  I realize this is solely personal taste.

Reply:
That sounds fine, but I think it would make sense to break up even further if so. For backwards compatibility, you can just `#include` the new files into this and existing includes will continue working

Reply:
new files included.